src/pages/join_info.tsx

```diff
@@ -1,18 +1,50 @@
 import React, { useState } from "react"; 
 import type { NextPage } from "next";
 import styled from "@emotion/styled"
+import Router from "next/router";
+import theme from "../styles/theme";
 
 const Join: NextPage = () => {
+  const [userNickname, setUserNickname] = useState<string>("");
+  const [checkButton, setCheckButton] = useState<boolean>(false); 
+
+  const handleNickname = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUserNickname(e.currentTarget.value);  
+  }
+
+  const handleButton = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault(); 
+    setCheckButton((current)=>!current); 
+  }
+  
+  const onSubmit = () => {
+    if(userNickname) {
+      Router.push("/"); 
+    } else {
+      alert("닉네임을 입력해주세요")
+    }
+  }
```

Comment:
p3: 함수 네이밍을 좀 더 구체화 하면 좋을거 같다는 생각이 듭니다.
밑에서 on Click 이벤트 발생시 onSubmit라는데 onHandleSubmit이나 뭔가 더 좋은 명시적인 이름이 있다면 그것을 props로 넣어주면 좋을 것 같습니다.

```suggestion
  const onSubmit = () => {
    if(userNickname) {
      Router.push("/"); 
    } else {
      alert("닉네임을 입력해주세요")
    }
  }
```

Reply:
=> handleOnSubmit